### PR TITLE
Profile based revisit

### PR DESCRIPTION
**Pull request by mkangia**
https://dimagi-dev.atlassian.net/browse/ICDS-674

Currently we [remove any latest enabled build profile setting](https://github.com/dimagi/commcare-hq/blob/31b15755fc727273bc4d5ec1e58491a4ed9404b6/corehq/apps/app_manager/views/releases.py#L627) when it's deactivated.
For https://dimagi-dev.atlassian.net/browse/ICDS-652, we need to update that logic and retain the settings made when its deactivated.
So i added a field `active` on the model and then set it to `False` when deactivating.
 
Took inspiration from `AppReleaseByLocation` as it's a similar (better?) model and modified `LatestEnabledBuildProfiles` to be like it, just moving the validations and status management to the model instead of the view.

There is an unrelated bug fix in [commit](https://github.com/dimagi/commcare-hq/pull/24663/commits/e2b6542b7ce99a9f96808bb0042e877012b28335)

FF: `RELEASE_BUILDS_PER_PROFILE`

buddies: @sravfeyn @calellowitz

### CODE REVIEW

File: corehq/apps/app_manager/views/releases.py [hunk 1]
```diff
@@ -593,7 +594,7 @@ def get(self, request, *args, **kwargs):
 
 @require_can_edit_apps
 def toggle_build_profile(request, domain, build_id, build_profile_id):
-    build = Application.get(build_id)
+    build = get_app_cached(request.domain, build_id)
```
Comment (mkangia) [inline corehq/apps/app_manager/views/releases.py]: related PR: https://github.com/dimagi/commcare-hq/pull/24664

**Comment by sravfeyn** [comment]
Since there is no context in the Jira ticket, curious why this is better behaviour than the previous?

**Comment by mkangia** [comment]
@sravfeyn the earlier implementation delete the record in case of deactivation while leads to a loss in history of changes. This PR updates the logic to mark a record as inactive instead of deleting it.
Let me know if i should explain more in-depth or if you want a walk through.

**Comment by sravfeyn** [comment]
Cool

**Review by esoergel — APPROVED** [review]
Looks good to me.  Only comment is that I'd recommend using the pattern of overriding the default model manager to exclude `active=False`.  Otherwise people will make usages that forget to exclude them and it's gonna be confusing when `active` is and isn't respected.  The `DomainLink` model has a nice example implementation:
https://github.com/dimagi/commcare-hq/blob/11d4443776dc33a24017f9cb398456b9df5b5099/corehq/apps/linked_domain/models.py#L41-L42

Nice PR structure too!

**Comment by mkangia** [comment]
@esoergel 
Thanks for the review Ethan.

> Nice PR structure too!
Great, thanks.

Umm.. about using the default filter. I think that's a good suggestion. I think it definitely makes sense for DomainLink because you would hardly ever want inactive links however in this case we do load all restrictions and not just enabled for the view(list) which is the main place where this would be used in HQ for web user. Actives are loaded when we are serving an app update request. So, both use cases (fetching all and fetching just active) are going to get equally used.
But considering its named `LatestEnabled` it should have all enabled ones but that's unfortunately because it was named according to the requirements earlier where we deleted records in case they were not needed.
I will think on this more and take in a followup PR, if needed.

**Comment by mkangia** [comment]
I merged an approved PR into this one, feel free to add comments if you didn't review on the follow up PR and have inputs.

**Pull request merged**